Composer/packages/client/src/recoilModel/dispatchers/orchestrator.ts

```diff
@@ -64,36 +73,46 @@ export const availableLanguageModels = (recognizerFiles: RecognizerFile[]) => {
         .filter((id) => id !== undefined)
         .some((lang) => Locales.map((l) => l.locale).includes(lang?.toLowerCase()))
     ) {
-      languageModels.push('multilang');
+      languageModels.push({
+        type: 'multilingual_intent',
+        name: botSettings?.orchestrator?.model?.multilingual_intent ?? 'default',
+      });
     }
   }
   return languageModels;
 };
 
 export const orchestratorDispatcher = () => {
-  const downloadLanguageModels = useRecoilCallback(({ snapshot }: CallbackInterface) => async (projectId: string) => {
-    const recognizers = await snapshot.getPromise(recognizersSelectorFamily(projectId));
-
-    const isUsingOrchestrator = recognizers.some(
-      ({ id, content }) => content.$kind === SDKKinds.OrchestratorRecognizer
-    );
-
-    if (isUsingOrchestrator) {
-      // Download Model Notification
-      const { addNotification, deleteNotification } = await snapshot.getPromise(dispatcherState);
-      const notification = createNotification(orchestratorDownloadNotificationProps());
-
-      try {
-        for (const languageModel of availableLanguageModels(recognizers)) {
-          await downloadModel('/orchestrator/download', languageModel, () => {
-            addNotification(notification);
-          });
+  const downloadLanguageModels = useRecoilCallback(
+    (callbackHelpers: CallbackInterface) => async (projectId: string) => {
+      const { snapshot } = callbackHelpers;
+
+      const recognizers = await snapshot.getPromise(recognizersSelectorFamily(projectId));
+
+      const isUsingOrchestrator = recognizers.some(({ content }) => content.$kind === SDKKinds.OrchestratorRecognizer);
+
+      if (isUsingOrchestrator) {
+        // Download Model Notification
+        const { addNotification, deleteNotification } = await snapshot.getPromise(dispatcherState);
+        const notification = createNotification(orchestratorDownloadNotificationProps());
+        const botSettings = await snapshot.getPromise(settingsState(projectId));
+
+        try {
+          for (const languageModel of availableLanguageModels(recognizers, botSettings)) {
+            await downloadModel('/orchestrator/download', languageModel, () => {
```

Comment:
shouldn't you add the notification first, then update it when needed?

Reply:
I'm confused as which notification Line 103 is updating

Reply:
Thanks @hatpick , I've changed the names so it's clearer - the update happens on the `downloadNotification`.